orderer/common/cluster/commauth_test.go

```diff
@@ -0,0 +1,431 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster_test
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/crypto"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/metrics"
+	"github.com/hyperledger/fabric/common/metrics/disabled"
+	comm_utils "github.com/hyperledger/fabric/internal/pkg/comm"
+	"github.com/hyperledger/fabric/internal/pkg/identity"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/cluster/mocks"
+	"github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type clusterAuthServer interface {
+	// Step passes an implementation-specific message to another cluster member.
+	Step(server orderer.ClusterNodeService_StepServer) error
+}
+
+type clusterServiceNode struct {
+	dialer       *cluster.PredicateDialer
+	handler      *mocks.Handler
+	nodeInfo     cluster.RemoteNode
+	srv          *comm_utils.GRPCServer
+	bindAddress  string
+	clientConfig comm_utils.ClientConfig
+	serverConfig comm_utils.ServerConfig
+	c            *cluster.Comm
+	dispatcher   clusterAuthServer
+}
+
+func (cn *clusterServiceNode) Step(stream orderer.ClusterNodeService_StepServer) error {
+	req, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+	if err == io.EOF {
+		return nil
+	}
+
+	i := rand.Int()
+	fmt.Println("Step function with execution id: ", i)
+
+	var channel string
+	if authReq := req.GetNodeAuthrequest(); authReq != nil {
+		fmt.Println("Auth request for channel: ", authReq.Channel, "with execution id", i)
+		channel = authReq.Channel
+		stream.Send(&orderer.ClusterNodeServiceStepResponse{
+			Payload: &orderer.ClusterNodeServiceStepResponse_TranorderRes{
+				TranorderRes: &orderer.TransactionOrderResponse{
+					Channel: authReq.Channel,
+					Status:  common.Status_SUCCESS,
+				},
+			},
+		})
+		req, err = stream.Recv()
+		if err != nil {
+			return err
+		}
+		if err == io.EOF {
+			return nil
+		}
+	}
+
+	fmt.Println("Tran/Consensus request for channel: ", channel, "with execution id", i)
+
+	if submitReq := req.GetNodeTranrequest(); submitReq != nil {
+		submitStepReq := &orderer.SubmitRequest{
+			Channel:           channel,
+			LastValidationSeq: submitReq.LastValidationSeq,
+			Payload:           submitReq.Payload,
+		}
+		return cn.c.DispatchSubmit(stream.Context(), submitStepReq)
```

Comment:
Why do you use the `c` reference? it's a `cluster.Comm`, you're testing here `cluster.ClusterService`